cmd/agent-smoke/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	smoke "github.com/grafana/agent/cmd/agent-smoke/internal"
+)
+
+func main() {
+	var (
+		debugLog          bool
+		namespace         string
+		kubeconfig        string
+		withTimeout       time.Duration
+		chaosFrequency    time.Duration
+		mutationFrequency time.Duration
+	)
+
+	flag.BoolVar(&debugLog, "debug", false, "enable debug logging")
+	flag.StringVar(&namespace, "namespace", "agent-smoke-test", "namespace smoke test should run in")
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.DurationVar(&withTimeout, "duration", time.Duration(0), "timeout after duration, example 3h")
+	flag.DurationVar(&chaosFrequency, "chaos-frequency", 30*time.Minute, "chaos frequency duration")
+	flag.DurationVar(&mutationFrequency, "mutation-frequency", 5*time.Minute, "mutation frequency duration")
+	flag.Parse()
+
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
+	level.AllowInfo()
+	level.Info(logger).Log("msg", "starting agent smoke framework")
+	if debugLog {
+		level.AllowDebug()
+		level.Debug(logger).Log("msg", "debug logging enabled")
+	}
```

Comment:
Can we use our existing log helper package here? I wouldn't be opposed to changing the arguments so it accepts a log level/format directly instead of a server config.  

https://github.com/grafana/agent/blob/5bf8cf452fa76c75387e30b6373630923679221c/pkg/util/logger.go#L28

(just calling `level.AllowInfo` doesn't actually do anything, it's an option you pass to `level.NewFilter`)

Reply:
Thanks for pointing this out, this makes the logging set up easier and requires less boiler plate. I didn't want to change the signature of our existing `NewLogger` API (although I imagine it probably isn't called in too many places), but opted to add a new `NewLoggerFromLevel` API. This allows me to take advantage of this existing helper package w/o having to change any other callers of this package. Let me know what you think.

Reply:
https://github.com/grafana/agent/pull/1225/commits/d05a28860042c0a38fca41b5c6e03778cf96f40b